crates/shared/src/tenderly_api.rs

```diff
@@ -0,0 +1,217 @@
+//! Module containing Tenderly API implementation.
+
+use crate::transport::extensions::StateOverrides;
+use anyhow::Result;
+use reqwest::{
+    header::{HeaderMap, HeaderValue},
+    Url,
+};
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+use web3::types::{H160, H256, U256};
+
+/// Trait for abstracting Tenderly API.
+#[async_trait::async_trait]
+pub trait TenderlyApi: Send + Sync + 'static {
+    async fn block_number(&self, network_id: &str) -> Result<u64>;
+    async fn simulate(&self, simulation: SimulationRequest) -> Result<SimulationResponse>;
+}
+
+const BASE_URL: &str = "https://api.tenderly.co/api";
+
+/// Tenderly HTTP API.
+pub struct TenderlyHttpApi {
+    url: Url,
+    client: reqwest::Client,
+    headers: HeaderMap,
+}
+
+impl TenderlyHttpApi {
+    /// Creates a new Tenderly API
+    pub fn new(client: reqwest::Client, user: &str, project: &str, api_key: &str) -> Result<Self> {
+        let mut headers = HeaderMap::new();
+        headers.insert("x-access-key", HeaderValue::from_str(api_key)?);
+
+        Ok(Self {
+            url: Url::parse(&format!("{BASE_URL}/v1/account/{user}/project/{project}/"))?,
+            client,
+            headers,
+        })
+    }
+
+    /// Creates a Tenderly API from the environment for testing.
+    pub fn test_from_env() -> Arc<dyn TenderlyApi> {
```

Comment:
Might as well use `#[cfg(test)]` here.

Reply:
Then we can't use this in the `orderbook` crate for example - where we make use of it.

Reply:
Makes sense. 👌